pycloudlib/gce/cloud.py

```diff
@@ -220,21 +220,16 @@ def launch(self, image_id, instance_type='n1-standard-1', user_data=None,
         ).execute()
         raise_on_error(operation)
 
-        self._wait_for_operation(operation, operation_type='zone')
```

Comment:
The call before this creates the instance, and the call after uses the information created to create our own instance object. If the first isn't finished, will the following run correctly?

Reply:
I believe so. Anecdotally, I've seen this succeed. More broadly, I think the lifecycle of the VM for an instance is decoupled from the lifecycle of the resource (in the RESTful-API sense) in GCE: https://cloud.google.com/compute/docs/reference/rest/v1/instances/insert says it "Creates an instance resource in the specified project using the data included in the request." I would expect the _resource_ creation to be complete when this call returns, including the ID assignment.

(Separately, the instance resource has multiple states in which it can be: https://cloud.google.com/compute/docs/instances/instance-life-cycle; these are represented within the resource.)